virtualmachine/vsphere/vm.go

```diff
@@ -726,3 +726,38 @@ func (vm *VM) GetSSH(options ssh.Options) (ssh.Client, error) {
 	client := ssh.SSHClient{Creds: &vm.Credentials, IP: ips[0], Port: 22, Options: options}
 	return &client, nil
 }
+
+func DeleteTemplate(vm *VM) error {
+	templates := make([]string, 0)
+	if err := SetupSession(vm); err != nil {
+		return err
+	}
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
+	}
+	for _, datastore := range vm.Datastores {
+		template := createTemplateName(vm.Template, datastore)
+		temp, err := findVM(vm, dcMo, template)
+		if err != nil {
+			templates = append(templates, template)
```

Comment:
I think template name is suppose to be unique. In that case, there won't be more than one match for template specified (to-be-deleted).
Do we need templates []string in that case?